sdks/python/apache_beam/runners/interactive/background_caching_job.py

```diff
@@ -87,11 +87,10 @@ def _should_end_condition_checker(self):
     return any([l.is_triggered() for l in self._limiters])
 
   def is_done(self):
-    is_terminated = self._pipeline_result.state is PipelineState.DONE
+    is_terminated = self._pipeline_result.state in (
+        PipelineState.DONE, PipelineState.CANCELLED)
```

Comment:
Ah this is odd indeed - this line must be hit by the test in capture_control_test.py, right? But codecov is marking it as not covered, correct?

cc: @udim

Reply:
Yeah, it definitely looks like it should be. Maybe Codev is trying to do something smarter like static analysis on the logic?

Reply:
Codecov reporting was buggy, should be better now. :)